mongo/options/findoptions.go

```diff
@@ -433,68 +433,89 @@ func MergeFindOneOptions(opts ...*FindOneOptions) *FindOneOptions {
 	return fo
 }
 
-// FindOneAndReplaceOptions represent all possible options to the FindOneAndReplace() function.
+// FindOneAndReplaceOptions represents options that can be used to configure a FindOneAndReplace instance.
 type FindOneAndReplaceOptions struct {
-	BypassDocumentValidation *bool           // If true, allows the write to opt out of document-level validation.
-	Collation                *Collation      // Specifies a collation to be used
-	MaxTime                  *time.Duration  // Specifies the maximum amount of time to allow the query to run.
-	Projection               interface{}     // Limits the fields returned for all documents.
-	ReturnDocument           *ReturnDocument // Specifies whether the original or updated document should be returned.
-	Sort                     interface{}     // Specifies the order in which to return results.
-	Upsert                   *bool           // If true, creates a a new document if no document matches the query.
+	// If true, writes executed as part of the operation will opt out of document-level validation on the server. This
+	// option is valid for MongoDB versions >= 3.2 and is ignored for previous server versions. The default value is
+	// false.
+	BypassDocumentValidation *bool
+
+	// Specifies a collation to use for string comparisons during the operation. This option is only valid for MongoDB
+	// versions >= 3.4. For previous server versions, the driver will return an error if this option is used. The
+	// default value is nil, which means the default collatioon of the collection will be used.
+	Collation *Collation
+
+	// The maximum amount of time that the query can run on the server. The default value is nil, meaning that there
+	// is no time limit for query execution.
+	MaxTime *time.Duration
+
+	// A document describing which fields will be included in the document returned by the operation. The default value
+	// is nil, which means all fields will be included.
+	Projection interface{}
+
+	// Specifies whether the original or replaced document should be returned by the operation. The default value is
+	// Before, which means the original document will be returned before the replacement is performed.
```

Comment:
returned before = returned from before

Otherwise it implies that the application gets the original document back before the server applies the replacement.

Reply:
Good catch. Done.